src/io/hdf/hdf_archive.cpp

```diff
@@ -161,22 +161,25 @@ bool hdf_archive::is_group(const std::string& aname)
     return false;
   hid_t p = group_id.empty() ? file_id : group_id.top();
   p       = (aname[0] == '/') ? file_id : p;
-  hid_t g = H5Gopen2(p, aname.c_str(), H5P_DEFAULT);
-  if (g < 0)
+  if (!(H5Lexists(p, aname.c_str(), H5P_DEFAULT) > 0))
```

Comment:
H5Lexists only query the existence of an entry. It doesn't verify if the entry is a group or not.
The function is_group needs to know if an entry is a group or not. This explains the failure in CI.
If an entry is a dataset, will H5Gopen2 still print the error stack?

Reply:
The Log VOL https://github.com/DataLib-ECP/vol-log-based does not support H5E.

If H5Gopen does not succeed, which would be the case for a dataset, it will dump the error stack.

I did not realize that the object can also be a dataset. In that case, it would probably be better to use H5Oget_info_by_name instead.

Reply:
I switch to using H5Oget_info_by_name in my test program:

(1) 336 Procs, 100000 groups

USING H5Gopen for a non-existent group: time is avg, min, max: 0.837418 0.796461 0.972710 s.
USING H5Oget_ for a non-existent group: time is avg, min, max:  0.800274 0.759203 0.854569 s.

(2) 336 Procs, 100000 groups

USING H5Gopen for an existent group: time is avg, min, max: 45.800670 15.395971 55.213545s
USING H5Oget_ (and H5Gopen) for an existent group: time is avg, min, max: 45.440913 14.367588 54.715002

Using a realistic number of 20 groups:

(1) 336 Procs, 20 groups

USING H5Gopen for a non-existent group: time is avg, min, max: 0.010792 0.002854 0.017069 s.
USING H5Oget_ for a non-existent group: time is avg, min, max: 0.009001 0.002897 0.015627 s.

(2) 336 Procs, 20 groups

USING H5Gopen for an existent group: time is avg, min, max: 0.025425 0.008726 0.042872 s.
USING H5Oget_ (and H5Gopen) for an existent group: time is avg, min, max:  0.026317 0.006041 0.041442 s.